### PR TITLE
New Disconnect Dialog

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request:

* Adds new, redesigned disconnect dialog
* Disconnect dialog now dynamically pulls in usage stats of Jetpack on the site
* Prep work for Survey Step in disconnect dialog
* Prep work for Plugins Install mobile

**Two Site Benefits - Mobile**
Screen Shot 2019-09-24 at 3 39 04 PM

**Three Site Benefits - Mobile**
Screen Shot 2019-09-24 at 3 39 04 PM

**Single Site Benefit - Wide**
Single Site Benefit - Wide

**Two Site Benefits - Wide**
Two Site Benefits - Wide

**Three Site Benefits - Wide**
Three Site Benefits - Wide

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Internal reference: parBdM-hs-p2

Redesign of Jetpack Disconnect Dialog 

#### Testing instructions:

1. Navigate to `/wp-admin/admin.php?page=jetpack#/dashboard`
2. Click "Manage Site Connection"
Screen Shot 2019-09-24 at 3 53 46 PM
3. Verify that your dialog roughly matches one of the above screenshot.
4. Modify the following code snippet and insert into `_inc/client/components/jetpack-termination-dialog/features.jsx` before the line `const siteBenefitCount = siteBenefits.length;` with varying amounts of additional site benefits to test the various cases.

```js
	siteBenefits = siteBenefits.concat( [
		{
			title: 'Brute Force Protection',
			description: 'The number of malicious login attempts blocked by Jetpack.',
			amount: 32030,
			gridIcon: 'lock',
		},
		{
			title: 'Contact Forms',
			description: 'The number of live Jetpack forms on your site right now.',
			amount: 31,
			gridIcon: 'align-image-center',
		},
		{
			title: 'Publicize',
			description: 'The number of live social media connections, powered by Jetpack.',
			amount: 3,
			gridIcon: 'share',
		},
		{
			title: 'Subscribers',
			description: 'The number of people subscribed to your updates through Jetpack.',
			amount: 4200,
			gridIcon: 'user',
		},
	] );
```

#### Proposed changelog entry for your changes:

Redesign of Jetpack Disconnect Dialog

### CODE REVIEW

**Comment by jeherve** [comment]
Removing the milestone since we discussed that this would be postponed to the next release:
p1568997915079200-slack-jetpack-crew

**Comment by cbauerman** [comment]
> Removing the milestone since we discussed that this would be postponed to the next release:
> p1568997915079200-slack-jetpack-crew

I managed to both flip the code freeze and release dates _and_ move Sep 23 to Oct 23 in my mind 🤦‍♂️ Thanks for the catch!

**Comment by scottsweb** [comment]
Thanks for the ping on this. Some initial thoughts:

* The close icon at the top right of the modal doesn't appear to be clickable (no cursor change) and it cannot be tabbed to either. So perhaps we just need to warp it in an `<a>`?
* We do not provide this modal when removing the plugin from the `plugins.php` screen yet. The same goes for the disconnect link on `admin.php?page=jetpack-debugger`. Guessing that might be coming soon along with the disconnect survey.
* On first load the modal was a little narrow, but it cleared up after subsequent refreshes, I think it might have been a build issue:

Screenshot 2019-09-25 at 15 22 37

* The responsiveness of the modal is great 👍 
* There were a couple of things in the original mockup (towards the bottom of the modal) that I think worked a little better:

Screenshot 2019-09-25 at 15 59 34

* The first CTA was changed to blue and "continue" to make it clear there is a second step
* The support link had a little more room around it
* The links were not italicised 
* We use a mixture of `log out` and `disconnect`, I am not sure where we landed on this in the end but it was certainly a point of discussion. On that same note it might be worth a copy review from @robertbpugh 

If I can help with anything then please let me know!

**Comment by cbauerman** [comment]
So, to wrap this whole conversation I need to add that this is designed to be PR 1 of 3 which explains some changes: 
1. ( **This PR** ) Redesigned Disconnect Dialog with 1-step, only on Jetpack Dashboard
2. 2-Step Disconnect Dialog with Survey
3. Add to Plugins Screen

So 
* Blue Continue Button -> Red Disconnect Button
* Missing from Plugins Screen
Are definitely missing by design.

I will be addressing/looking into:
* Cursor on modal close
* Link italics
* Support link space

and I am looking for feedback on the copy. My plan is to use "Disconnect Jetpack" on the dashboard and "Disable Jetpack" on the upcoming plugins screen. ( cc @robertbpugh )

let me know how that sounds @scottsweb!

**Comment by cbauerman** [comment]
Did not mean to close, pressed the button without looking to close 🤦‍♂️

**Comment by scottsweb** [comment]
That all sounds good to me, it is good to break this down into separate PRs. Your copy proposal makes sense too given that the outcomes are slightly different based on the where the modal is initiated from.

**Comment by scottsweb** [comment]
Did you want me to test this again @cbauerman ?

**Comment by cbauerman** [comment]
> Did you want me to test this again @cbauerman now the styles have been updated ?

@scottsweb yes please!

**Comment by scottsweb** [comment]
Tested and looking good, giving the design 👍

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/features.jsx]: "several" is not perfect when you're only using a couple of modules. Do you think we could reword that, or make that part of the string dynamic?

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/features.jsx]: I wonder if each reason could be extracted to keep the final `return` shorter and easier to parse?

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/dialog.jsx]: Since deactivating the plugin will definitely happen, should we remove the parenthesis?

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/dialog.jsx]: We use `h2` headings in other areas (like the Mobile login modal next to this one). Maybe we should use `h2` here as well.

File: _inc/client/components/data/query-site-benefits/index.jsx
Comment (jeherve) [inline _inc/client/components/data/query-site-benefits/index.jsx]: Could you clarify why we need to bring in `isDevMode` here? The modal will not be displayed anywhere when in development mode, right?

**Review by jeherve — CHANGES_REQUESTED** [review]
I have a few comments and questions below.

I am also not sure if this is caused by my screen size, but it felt odd to have a bottom margin, but no space at the top:
image

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/features.jsx]: cc @robertbpugh

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/dialog.jsx]: 🤦‍♂️

File: _inc/client/components/jetpack-termination-dialog/style.scss
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/style.scss]: cc @scottsweb

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/dialog.jsx]: imo it looked pretty weird with the little one, which is why I boosted the size.

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/dialog.jsx]: sounds good 👍

File: _inc/client/components/data/query-site-benefits/index.jsx
Comment (cbauerman) [reply _inc/client/components/data/query-site-benefits/index.jsx]: leftover from other changes, I will get rid of it

File: _inc/client/components/data/query-site-benefits/index.jsx
Comment (simison) [inline _inc/client/components/data/query-site-benefits/index.jsx]: Could this be done in some other way than introducing legacy methods in a new code? Otherwise, you're just punting this work for anyone who attempts to update React in the future.

https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

File: _inc/client/components/jetpack-termination-dialog/style.scss
Comment (simison) [inline _inc/client/components/jetpack-termination-dialog/style.scss]: Would it be possible to use SASS variable here?

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (simison) [inline _inc/client/components/jetpack-termination-dialog/features.jsx]: Should the whole disconnect modal be hidden for Atomic sites instead? 🤔

File: _inc/client/components/data/query-site-benefits/index.jsx
Comment (cbauerman) [reply _inc/client/components/data/query-site-benefits/index.jsx]: I mindlessly copied this component in from another PR 🤦‍♂️. I will avoid having a PR with `UNSAFE_x` in it 👍

**Comment by jeherve** [comment]
@cbauerman What's the status on this? Did you have the chance to address all feedback above?

**Comment by cbauerman** [comment]
> @cbauerman What's the status on this? Did you have the chance to address all feedback above?

We are at about 80%  of the feedback addressed. I will push myself to get this ready for re-review by EoD today.

**Comment by cbauerman** [comment]
> > @cbauerman What's the status on this? Did you have the chance to address all feedback above?
> 
> We are at about 80% of the feedback addressed. I will push myself to get this ready for re-review by EoD today.

@jeherve I over-committed today. Would End-of-Week at worst work?

**Comment by jeherve** [comment]
Yup, all good!

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/features.jsx]: how does "Send a question to the Jetpack support team." sound?

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/dialog.jsx]: it turns out blogId is automatically included here, so I am removing siteName

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/features.jsx]: going to "Jetpack is powering features on your site" to be more generic, there are a lot of cases going on.

File: _inc/client/components/jetpack-termination-dialog/features.jsx
Comment (jeherve) [reply _inc/client/components/jetpack-termination-dialog/features.jsx]: I like it. 👍

File: _inc/client/components/jetpack-termination-dialog/single-feature.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/single-feature.jsx]: I think we'll need this amount to be internationalized, maybe, so you get something like `2,000` instead of `2000`.

Something like this should do the trick:

```diff
diff --git a/_inc/client/components/jetpack-termination-dialog/single-feature.jsx b/_inc/client/components/jetpack-termination-dialog/single-feature.jsx
index 4b2697e6e..c950164bc 100644
--- a/_inc/client/components/jetpack-termination-dialog/single-feature.jsx
+++ b/_inc/client/components/jetpack-termination-dialog/single-feature.jsx
@@ -2,6 +2,7 @@
  * External dependencies
  */
 import Gridicon from 'components/gridicon';
+import { numberFormat } from 'i18n-calypso';
 import React from 'react';
 
 const SingleFeature = ( { amount, description, gridIcon, title } ) => {
@@ -12,7 +13,7 @@ const SingleFeature = ( { amount, description, gridIcon, title } ) => {
 				<Gridicon icon={ gridIcon } />
 			</div>
 			<div className="jetpack-termination-dialog__feature-body">
-				<p className="jetpack-termination-dialog__feature-body-amount">{ amount }</p>
+				<p className="jetpack-termination-dialog__feature-body-amount">{ numberFormat( amount ) }</p>
 				<p className="jetpack-termination-dialog__feature-body-description">{ description }</p>
 			</div>
 		</div>

```

**Before**

image


**After**

image

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/dialog.jsx]: Should this be 

```suggestion
		analytics.tracks.recordEvent( 'jetpack_termination_dialog_view', {
```

here and below, or is the double `termination` here on purpose?

Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/dialog.jsx]: We're effectively tracking this twice now. Clicking on the "Manage site connection" link triggers this new event you've added, as well as the existing one, `jetpack_wpa_click` with props `manage_site_connection`. Do we need both?

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (jeherve) [inline _inc/client/components/jetpack-termination-dialog/dialog.jsx]: `this.props.terminateJetpack()` does not exist yet.

**Review by jeherve — CHANGES_REQUESTED** [review]

File: _inc/client/components/jetpack-termination-dialog/dialog.jsx
Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/dialog.jsx]: 'tis a typo 🤦‍♂

Comment (cbauerman) [reply _inc/client/components/jetpack-termination-dialog/dialog.jsx]: we can strike it for now, it is a type and should be `jetpack_termination_dialog_features_view`, and is only relevant once both pieces ( the features and the survey ) are up and running.

**Review by jeherve — DISMISSED** [review]
This works well for me now. I think the the disconnect / terminate differences could be a bit confusing for someone taking a first look at that code, but that can be addressed in a future PR imo.

**Review by jeherve — APPROVED** [review]
Re-approving after a rebase. I'll merge this now to avoid any more conflicts with `master` in the future.

**Pull request merged**
